gsql/cache/cache_key.py

```diff
@@ -0,0 +1,74 @@
+from functools import lru_cache
+from key_pair import Keypair
+from gsql.cache.cache_handler import (
+    CacheDict,
+    check_json_file_exists,
+    count_lines_in_json,
+    create_json_file,
+    keys_exists,
+    read_from_json,
+    remove_key_from_json,
+    write_to_json,
+)
+from gsql.cache.constant import MAXIMUM_SIZE_OF_JSON_FILE
+
+
+"""
+    Cachekey contains the type of query,
+    the query itself, database id and the sheet id.
+"""
+
+
+class Cachekey:
+    def __init__(self, query_type, query, db_id, sheet_id) -> None:
+        self.query_type = query_type
+        self.query = query
+        self.db_id = db_id
+        self.sheet_id = sheet_id
+
+    @lru_cache(maxsize=10000)
+    def validate(self):
```

Comment:
lru_cache caches the arguments passed to a function, it won't actually work here, please find another way to cache the queries. 

Ref - [StackOverflow](https://stackoverflow.com/questions/49883177/how-does-lru-cache-from-functools-work)